tests/observatory/dags/telescopes/test_orcid.py

```diff
@@ -0,0 +1,487 @@
+        self.project_id = os.getenv('TEST_GCP_PROJECT_ID')
+        self.data_location = os.getenv('TEST_GCP_DATA_LOCATION')
+
+        self.first_execution_date = pendulum.datetime(year=2021, month=2, day=1)
+        self.second_execution_date = pendulum.datetime(year=2021, month=3, day=1)
+
+        # orcid records
+        self.records = {}
+        for file in ['0000-0002-9227-8610.xml', '0000-0002-9228-8514.xml', '0000-0002-9229-8514.xml']:
+            self.records[file] = {'blob': f"{file[-7:-4]}/{file}",
+                                  'path': os.path.join(test_fixtures_path("telescopes", "orcid"), file)}
+        # last modified file
+        self.last_modified_path = os.path.join(test_fixtures_path("telescopes", "orcid"), 'last_modified.csv.tar')
+
+        # release used for method tests
+        self.release = OrcidRelease(OrcidTelescope.DAG_ID, datetime(2020, 1, 1), datetime(2020, 2, 1), False,
+                                    max_processes=1)
+
+    def test_dag_structure(self):
+        """ Test that the ORCID DAG has the correct structure.
+        :return: None
+        """
+
+        dag = OrcidTelescope().make_dag()
+        self.assert_dag_structure({
+            'check_dependencies': ['get_release_info'],
+            'get_release_info': ['transfer'],
+            'transfer': ['download_transferred'],
+            'download_transferred': ['transform'],
+            'transform': ['upload_transformed'],
+            'upload_transformed': ['bq_load_partition'],
+            'bq_load_partition': ['bq_delete_old'],
+            'bq_delete_old': ['bq_append_new'],
+            'bq_append_new': ['cleanup'],
+            'cleanup': []
+        }, dag)
+
+    def test_dag_load(self):
+        """ Test that the ORCID DAG can be loaded from a DAG bag.
+        :return: None
+        """
+
+        with ObservatoryEnvironment().create():
+            dag_file = os.path.join(module_file_path('observatory.dags.dags'), 'orcid.py')
+            self.assert_dag_load('orcid', dag_file)
+
+    @patch('observatory.dags.telescopes.orcid.aws_to_google_cloud_storage_transfer')
+    @patch("observatory.dags.telescopes.orcid.boto3.client")
+    def test_telescope(self, mock_client, mock_transfer):
+        """ Test the ORCID telescope end to end.
+        :return: None.
+        """
+        # Setup Observatory environment
+        env = ObservatoryEnvironment(self.project_id, self.data_location)
+        dataset_id = env.add_dataset()
+
+        # Set up google cloud sync bucket
+        orcid_bucket = env.add_bucket()
+
+        # Setup Telescope
+        telescope = OrcidTelescope(dataset_id=dataset_id)
+        dag = telescope.make_dag()
+
+        # Create the Observatory environment and run tests
+        with env.create():
+            # Add connection
+            conn = Connection(conn_id=AirflowConns.ORCID,
+                              uri="aws://UWLA41aAhdja:AJLD91saAJSKAL0AjAhkaka@")
```

Comment:
I presume that these are not real credentials?

Reply:
No they are not :-) I generated them with canarytoken, just trying that out, so you get a notification when someone tries to use the token. It does not really serve any purpose for us though, because they are publicly available obviously. The idea is you can use the tokens to check whether there is a security fault somewhere, if you expect the token to be kept private.

Reply:
Ah I see, that sounds helpful!